### ISSUE TITLE
L064 rule false positive for quoted identifiers in expressions

### ISSUE THREAD

**Issue by KulykDmytro** [sqlfluff/sqlfluff#3186]
### What Happened

in case of usage quoted identifiers like (`date`, `timestamp` etc) as a part of expression `L064` rule caused false-positive violation

### Observed Behaviour

```verilog
L:  70 | P:  37 | L064 | Inconsistent use of preferred quote style 'single quotes'. Use 'date' instead of "date".
```

### How to reproduce

```sql
    select
        player_id,
        risk_level_id,
        risk_level,
        cast(from_iso8601_timestamp("date") as timestamp) as event_date
    from raw.player_limits
    where dt < date'{{ ds }}' + interval '1' day
```

* Dialect: hive
* Version: 0.13.0

### Configuration
```ini
[sqlfluff]
templater = jinja
output_line_length = 180
exclude_rules = L022,L031,L032,L033,L034,L042,L051
dialect = hive

[sqlfluff:rules]
max_line_length = 120

[sqlfluff:rules:L028]
single_table_references = unqualified

[sqlfluff:rules:L029]
# quoted_identifiers_policy = none
ignore_words =
    position,type,name,language,state,size,source,resource,condition,format,account,comment,result,operator,data,
    system,domain,location,role,alias,template,count,rank,date

[sqlfluff:templater:jinja]
library_path = operation/deploy/lint
apply_dbt_builtins = false

[sqlfluff:templater:jinja:context]
# TODO: move to lint-mock module
ds = 2021-11-11
```

**Comment by tunetheweb** [comment]
Can you explain why this is a false positive @KulykDmytro ?

This is using double quotes: `from_iso8601_timestamp("date")`
This is using single quote: `date'{{ ds }}' + interval '1' day`

And the rule identifies inconsistencies.

Is it not possible to make these consistent?

**Comment by KulykDmytro** [comment]
this is why:
`"date"` - is a column name
`date'{{ ds }}'` - is a literal constant

**Comment by tunetheweb** [comment]
OK so the issue is we're parsing this in two parts:

```
[L:  7, P: 16]      |                    keyword:                                  'date'
[L:  7, P: 20]      |                    literal:                                  "''"
```

Instead of as one literal. Gotcha.

**Comment by KulykDmytro** [comment]
in this case `date'2022-02-11'` is a date literal "constructor"

**Comment by KulykDmytro** [comment]
issue not fixed

**Comment by barrywhart** [comment]
@KulykDmytro, @tunetheweb: Are we saying that in Hive, double quotes can be part of the column name, rather than **delimiters** for a column name? If so, I don't see how SQLFluff can possibly know the correct thing to do. Should we disable this (and possibly other rules such as L059) for Hive? We did something similar for Snowflake a few days ago (PR #3260).

**Comment by barrywhart** [comment]
Closing due to no response.
